runtime/type-registry/src/lib.rs

```diff
@@ -1,16 +1,134 @@
-use std::collections::HashMap;
-use std::fmt;
-use std::fmt::Debug;
-use std::fmt::Display;
-use std::fmt::Formatter;
-use std::ptr;
-use std::sync::LazyLock;
-use std::sync::RwLock;
+#![no_std]
+/* use the C allocator; don't use libstd */
+extern crate alloc;
+use libc_alloc::LibcAlloc;
+
+#[global_allocator]
+static ALLOCATOR: LibcAlloc = LibcAlloc;
+
+struct StdErrWriter;
+impl alloc::fmt::Write for StdErrWriter {
+    fn write_str(&mut self, s: &str) -> Result<(), core::fmt::Error> {
+        unsafe extern "C" {
+            unsafe fn write(fd: i32, buf: *const u8, len: usize);
+        }
+        unsafe {
+            write(2, s.as_ptr(), s.len());
+            write(2, "\n".as_ptr(), 1);
```

Comment:
Why are we writing a newline here?

Reply:
Good catch, the newline should be in the caller (where we can just use `writeln!` instead of `write!`). Fixed.